Add periodic vSphere IPI job for 4.5/4.6 testgrid

enxebre opened the pull request:
This is to add a vSphere IPI e2e test to the periodic jobs for 4.5/4.6 tagged with `ci.openshift.io/release-type: informing` and follow up with a PR against https://github.com/kubernetes/test-infra to include them in the test grid dashboard.

I copied the spec from https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/installer/openshift-installer-release-4.5-presubmits.yaml#L946-L1009
/hold
As I'm not sure about this syntax.

enxebre commented:
cc @bparees @petr-muller

bparees commented:
> I copied the spec from https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/installer/openshift-installer-release-4.5-presubmits.yaml#L946-L1009

you should not be editing this file manually, you should be generating these pages using the testgrid-config-generetor (after changing the openshift job metadata as needed to mark the jobs informing or blocking):
https://github.com/openshift/ci-tools/tree/master/cmd/testgrid-config-generator

enxebre commented:
>you should not be editing this file manually, you should be generating these pages using the testgrid-config-generetor (after changing the openshift job metadata as needed to mark the jobs informing or blocking):
https://github.com/openshift/ci-tools/tree/master/cmd/testgrid-config-generator

Based on the readme on https://github.com/openshift/ci-tools/tree/master/cmd/testgrid-config-generator I thought that was to generate the code against the https://github.com/kubernetes/test-infra repo.
This PR is trying to do create the new jobs tagged with `ci.openshift.io/release-type: informing` (analogous to https://github.com/openshift/release/pull/8465/files) so then I can run `testgrid-config-generator` to PR against https://github.com/kubernetes/test-infra ?

bparees commented:
> @bparees Based on the readme on https://github.com/openshift/ci-tools/tree/master/cmd/testgrid-config-generator I thought that was to generate the code against the https://github.com/kubernetes/test-infra repo.

yeah sorry, i was insufficiently caffeinated when i looked at this, i thought it was a test-infra PR.

bparees commented on ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:
```diff
@@ -437,6 +437,77 @@ periodics:
+    ci-operator.openshift.io/prowgen-controlled: "true"
+    ci-operator.openshift.io/variant: ocp-4.5
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-release-master-ocp-4.5-e2e-vsphere
```
i think you should only be adding the 4.6 variant to "master"

bparees replied:
(I guess the generator did this?)

enxebre replied:
yeh this was generated by `make jobs`.

Then I manually created the ones in `openshift-release-release-4.6-periodics.yaml` and `openshift-release-release-4.5-periodics.yaml` as I understand those are the one to be exposed to the test grid and there's no automated generation support yet, but they complain with `error: invalid configuration: you must define at least one test or image build in ‘tests’ or ‘images’`

jcpowermac commented:
@enxebre I _think_ you need the `upi-installer` in the config
I couldn't get the job to run without it.
https://github.com/openshift/release/pull/9204/files#diff-b7728c9d669b573babd14a6224047457R10

enxebre commented:
I dropped the manual jobs I had added into openshift-release-release-4.5-periodics.yaml/4.6 and added the annotations `ci.openshift.io/release-type` to the ones autogenerated in openshift-release-master-periodics.yaml but that way prow/config complains.

jcpowermac commented:
@enxebre this can be closed. We have periodics for upi and ipi using multi-step (or whatever they are calling).
See: https://prow.ci.openshift.org/?type=periodic&job=*vsphere*

jcpowermac commented:
/close
